File: leet-code/postorderTraversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var postorderTraversal = function (root) {
  let res = []
  if (root == null) {
    return res
  }
  let fn = function (root) {
    if (root.left) {
      fn(root.left)
    }
    if (root.right) {
      fn(root.right)
    }
    res.push(root.val)
  }
  fn(root)
  return res
}

File: leet-code/postorderTraversal.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func postorderTraversal(root *TreeNode) []int {
	res := make([]int, 0)
	var fn func(root *TreeNode)
	if root == nil {
		return res
	}
	fn = func(root *TreeNode) {
		if root.Left != nil {
			fn(root.Left)
		}
		if root.Right != nil {
			fn(root.Right)
		}
		res = append(res, root.Val)
	}
	fn(root)
	return res
}
